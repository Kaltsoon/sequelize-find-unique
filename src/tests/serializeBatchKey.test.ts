import { Op } from 'sequelize';

import serializeBatchKey from '../serializeBatchKey';
import { sequelize, Comment } from './models';

describe('serializeBatchKey', () => {
  test('serializes primitive values correctly', () => {
    expect(
      serializeBatchKey({
        where: {
          firstName: 'Kalle',
          lastName: 'Ilves',
        },
        attributes: ['firstName', 'lastName'],
      }),
    ).toMatchSnapshot();
  });

  test('serializes model references correctly', () => {
    expect(
      serializeBatchKey({
        where: {
          firstName: 'Kalle',
          lastName: 'Ilves',
        },
        attributes: ['firstName', 'lastName'],
        include: [{ model: Comment, as: 'comments' }],
      }),
    ).toMatchSnapshot();
  });

  test('serializes fn correctly', () => {
    expect(
      serializeBatchKey({
        where: {
          firstName: 'Kalle',
          lastName: 'Ilves',
        },
        attributes: [
          'firstName',
          'lastName',
          [sequelize.fn('COUNT', sequelize.col('column')), 'column_alias'],
        ],
      }),
    ).toMatchSnapshot();
  });

  test('serializes Op correctly', () => {
    expect(
      serializeBatchKey({
        where: {
          firstName: 'Kalle',
          lastName: 'Ilves',
        },
        attributes: ['firstName', 'lastName'],
        include: [
          {
            model: Comment,
            as: 'comments',
            where: {
              [Op.or]: [{ id: { [Op.gt]: 5 } }, { id: { [Op.ne]: 100 } }],
            },
          },
        ],
      }),
    ).toMatchSnapshot();
  });

  test('throws error for non-serializable values', () => {
    expect(() =>
      serializeBatchKey({
        where: {
          firstName: 'Kalle',
          lastName: 'Ilves',
        },
        attributes: [
          'firstName',
          'lastName',
          () => {
            return undefined;
          },
        ] as any,
      }),
    ).toThrowErrorMatchingSnapshot();
  });
});
