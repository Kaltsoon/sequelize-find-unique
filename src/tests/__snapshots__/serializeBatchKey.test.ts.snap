// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializeBatchKey serializes Op correctly 1`] = `"{\\"attributes\\":[\\"firstName\\",\\"lastName\\"],\\"include\\":[{\\"as\\":\\"comments\\",\\"model\\":\\"__model:comments\\",\\"where\\":{\\"__op:Symbol(or)\\":[{\\"id\\":{\\"__op:Symbol(gt)\\":5}},{\\"id\\":{\\"__op:Symbol(ne)\\":100}}]}}],\\"where\\":[\\"firstName\\",\\"lastName\\"]}"`;

exports[`serializeBatchKey serializes fn correctly 1`] = `"{\\"attributes\\":[\\"firstName\\",\\"lastName\\",[{\\"args\\":[{\\"col\\":\\"column\\"}],\\"fn\\":\\"COUNT\\"},\\"column_alias\\"]],\\"where\\":[\\"firstName\\",\\"lastName\\"]}"`;

exports[`serializeBatchKey serializes model references correctly 1`] = `"{\\"attributes\\":[\\"firstName\\",\\"lastName\\"],\\"include\\":[{\\"as\\":\\"comments\\",\\"model\\":\\"__model:comments\\"}],\\"where\\":[\\"firstName\\",\\"lastName\\"]}"`;

exports[`serializeBatchKey serializes primitive values correctly 1`] = `"{\\"attributes\\":[\\"firstName\\",\\"lastName\\"],\\"where\\":[\\"firstName\\",\\"lastName\\"]}"`;

exports[`serializeBatchKey throws error for non-serializable values 1`] = `"Found non-serializable function from findUnique options"`;
